docs/developer/new-contributor-projects.html

```diff
@@ -1372,37 +1373,117 @@ <h2 id="javaparser-to-javac-parse">Replace JavaParser by javac</h2>
+  package <a href="https://checkerframework.org/api/org/checkerframework/framework/ajava/package-summary.html"><code>org.checkerframework.framework.ajava</code></a>.
+  Currently,
+  the <a href="https://checkerframework.org/api/org/checkerframework/framework/ajava/package-summary.html"><code>org.checkerframework.framework.ajava</code></a>
+  package
+  is used by <a href="https://checkerframework.org/api/org/checkerframework/common/basetype/BaseTypeVisitor.html"><code>BaseTypeVisitor</code></a>,
+ <a href="https://checkerframework.org/api/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.html"><code>WholeProgramInferenceJavaParserStorage</code></a>,
+ and
+ <a href="https://checkerframework.org/api/org/checkerframework/framework/stub/AnnotationFileParser.html"><code>AnnotationFileParser</code></a>.
+</p>
 
+<p>The concrete tasks are:</p>
+<ul>
+  <li>For each class in
+  <a href="https://checkerframework.org/api/org/checkerframework/framework/ajava/package-summary.html"><code>org.checkerframework.framework.ajava</code></a>,
+  create a new version that uses
+  the <a href="https://github.com/plume-lib/javac-parse">javac-parse</a>
+  library instead of JavaParser.
+
+  For example,
+  <a href="https://checkerframework.org/api/org/checkerframework/framework/ajava/AnnotationEqualityVisitor.html"><code>AnnotationEqualityVisitor</code></a>
+  operates over two JavaParser trees, and you need to create a variant that
+  operates over two javac trees.
+  </li>
+  <li>
+    Convert as many clients as possible to use the new classes.
+  </li>
+  <li>
+    After all tests pass, remove as many old classes as possible.
+  </li>
+</ul>
 
 
+<h3 id="javacparse-annotated-jdk">Replace JavaParser for the annotated JDK</h3>
+
+<p>
+  The Checker Framework distribution contains a minimized copy of the
+  annotated JDK, where each file has been changed from a .java file to a
+  stub file.  This process is performed by
+  class <code>JavaStubifier</code>.  These stub files are not readable by
+  javac's parser (see the <a href="#javaparser-stub">next subproject</a>
+  for an explanation).  To use javac's parser, the distributed annotated
+  JDK must contain legal Java files.
+</p>
+
+<p>
+  This subproject has these goals:
+</p>
+<ul>
+  <li>Create an <code>Ajavaifier</code> program that is a modified version
+  of <code>JavaStubifier</code>.  <code>Ajavaifier</code> differs in two
+  ways.
+    <ul>
+      <li>It parses the original annotated JDK using javac's parser
+        (via <a href="https://github.com/plume-lib/javac-parse">javac-parse</a>)
+        rather than using JavaParser.
+      </li>
+      <li>
+        It produces minimized Java files rather than producing stub files.
+        These will be slightly larger than the current stub files.  For
+        example, method bodies in the input will be replaced by <code>{ throw
+          new Error(); }</code> rather than by <code>;</code>.
```

Comment:
Would these then be `.ajava` files rather than stub files?  (Not a blocking comment)